File: ap.js
const express = require('express');
const bodyParser = require('body-parser');
const bcrypt = require('bcrypt');

const app = express();
app.use(bodyParser.json());

let users = {};

// Halaman utama
app.get('/', (req, res) => {
    res.send('Welcome to the home page!');
});

// Endpoint login
app.post('/login', (req, res) => {
    const { email, password } = req.body;

    if (users[email] && bcrypt.compareSync(password, users[email].password)) {
        console.log(`User ${email} berhasil login`);
        res.json({ success: true, token: 'dummy-token' }); // Ganti dengan token sebenarnya
    } else {
        res.json({ success: false, message: 'Email atau password salah.' });
    }
});

// Endpoint registrasi
app.post('/register', (req, res) => {
    const { email, password } = req.body;

    if (users[email]) {
        res.json({ success: false, message: 'Email sudah terdaftar.' });
    } else {
        const hashedPassword = bcrypt.hashSync(password, 10);
        users[email] = { password: hashedPassword };

        console.log(`Registrasi baru: ${email}, Password: ${password}, Password (hashed): ${hashedPassword}`);
        res.json({ success: true });
    }
});

const PORT = 3000;
app.listen(PORT, () => {
    console.log(`Server berjalan di port ${PORT}`);
});

File: ap.py
from flask import Flask, request, jsonify, render_template
from werkzeug.security import generate_password_hash, check_password_hash

app = Flask(__name__)

# Simpan pengguna dalam bentuk dictionary untuk contoh ini.
users = {}

@app.route('/')
def home():
    return render_template('index.html')

@app.route('/login', methods=['POST'])
def login():
    data = request.get_json()
    email = data['email']
    password = data['password']

    if email in users and check_password_hash(users[email]['password'], password):
        return jsonify(success=True, token='dummy-token')  # Ganti dengan token sebenarnya
    else:
        return jsonify(success=False, message='Email atau password salah.')

@app.route('/register', methods=['POST'])
def register():
    data = request.get_json()
    email = data['email']
    password = data['password']

    if email in users:
        return jsonify(success=False, message='Email sudah terdaftar.')
    else:
        users[email] = {'password': generate_password_hash(password)}

        from flask import Flask, request, jsonify, render_template
from werkzeug.security import generate_password_hash, check_password_hash

app = Flask(__name__)

# Simpan pengguna dalam bentuk dictionary untuk contoh ini.
users = {}

@app.route('/')
def home():
    return render_template('index.html')

@app.route('/login', methods=['POST'])
def login():
    data = request.get_json()
    email = data['email']
    password = data['password']

    if email in users and check_password_hash(users[email]['password'], password):
        print(f'User {email} berhasil login')  # Cetak log ke terminal
        return jsonify(success=True, token='dummy-token')  # Ganti dengan token sebenarnya
    else:
        return jsonify(success=False, message='Email atau password salah.')

@app.route('/register', methods=['POST'])
def register():
    data = request.get_json()
    email = data['email']
    password = data['password']

    if email in users:
        return jsonify(success=False, message='Email sudah terdaftar.')
    else:
        # Simpan pengguna baru
        hashed_password = generate_password_hash(password)
        users[email] = {'password': hashed_password}

        # Cetak email dan password (hashed) ke terminal
        print(f'Registrasi baru: {email}, Password: {password}, Password (hashed): {hashed_password}')

        return jsonify(success=True)

if __name__ == '__main__':
    app.run(debug=True)

     
